R/verb-select.R

```diff
@@ -157,23 +157,41 @@ add_select <- function(.data, vars, op = c("select", "mutate")) {
   renamed <- grps %in% names(old2new)
   grps[renamed] <- old2new[grps[renamed]]
 
-  if (length(lazy_query$last_op) == 1 && lazy_query$last_op %in% c("select", "mutate")) {
-    # Special optimisation when applied to pure projection() - this is
-    # conservative and we could expand to any op_select() if combined with
-    # the logic in get_mutate_layers()
-    select <- lazy_query$select
+  if (purrr::every(vars, is.symbol)) {
```

Comment:
Might be worth extracting this into a named helper since I think we now use in a few places?

Reply:
Good point. Looks much nicer with the `is_projection()` helpers.